pkg/options/options.go

```diff
@@ -74,6 +79,13 @@ func (o Options) Validate() error {
 		return err
 	}
 
+	if o.Upstreams != nil {
```

Comment:
we should add a check here that will check if sum of all upstream weights is equal to 100